### PR TITLE
kvcoord: Keep track of the number of local rangefeeds

### LINKED ISSUE #114187: kvcoord: Export metric on number of local rangefeeds
Publish number of local rangefeeds running.  These use local RPC bypass mechanism,
and various system characteristics (memory, etc) depend on how many local/remote ranges node connects
to

(FYI: @stevendanna )

Jira issue: CRDB-33386

### PR DESCRIPTION
**Pull request by miretskiy**
Keep track of the number of rangefeed streams established to local node.  Local node streams use different RPC mechanism (rpc bypass); it is helpful to know how many local vs remote streams are established by rangefeed.

Fixes #114187

Release note: None

### CODE REVIEW

File: pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go [hunk 1]
```diff
@@ -415,6 +419,22 @@ func (a *activeRangeFeed) onRangeEvent(
...
 }
 
+// onConnect is a callback invoked when attempt to connect to the specified
+// destination is made.
+func (a *activeRangeFeed) onConnect(
+	dest rpc.RestrictedInternalClient, metrics *DistSenderRangeFeedMetrics,
+) {
+	if a.onDisconnect != nil {
+		a.onDisconnect()
+	}
```
Comment (erikgrinaker) [inline pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:429]: Why are we calling `onDisconnect()` during `onConnect()`?

Comment (miretskiy) [reply pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:429]: Perhaps I need better name.. Basically, if onConnect() failed and we are trying to connect to another replica,
we need to "disconnect" -- i.e. update metrics to decrement them.

Comment (erikgrinaker) [reply pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go:429]: Isn't that an artifact of either calling `onConnect()` prematurely (before we know whether the call succeded), or not calling `onDisconnect()` when we lose a connection?

Comment (miretskiy) [reply pkg/kv/kvclient/kvcoord/dist_sender_rangefeed.go]: Ptal.  Dropped onDisconnnect.  Just a simple bool now.

**Review by erikgrinaker — APPROVED** [review]

**Comment by miretskiy** [comment]
Bors r+

**Pull request merged**